src/main/kotlin/org/jitsi/videobridge/ice/IceConfig.kt

```diff
@@ -0,0 +1,185 @@
+import org.jitsi.utils.config.SimpleProperty
+import org.jitsi.utils.config.helpers.attributes
+import java.util.Objects
+
+class IceConfig {
+    class Config {
+        companion object {
+            /**
+             * The property which enables ICE/TCP.
+             */
+            class TcpEnabledProperty : FallbackProperty<Boolean>(
+                legacyConfigAttributes {
+                    name("org.jitsi.videobridge.DISABLE_TCP_HARVESTER")
+                    readOnce()
+                    // The old property is named 'disable', while the new one
+                    // is 'enable', so invert the old value
+                    transformedBy { !it }
+                },
+                newConfigAttributes {
+                    name("videobridge.ice.tcp.enabled")
+                    readOnce()
+                }
+            )
+            private val tcpEnabledProp = TcpEnabledProperty()
+
+            @JvmStatic
+            fun tcpEnabled() = tcpEnabledProp.value
+
+            /**
+             * The property which configures the ICE/TCP port.
+             */
+            class TcpPortProperty : LegacyFallbackConfigProperty<Int>(
+                Int::class,
+                readOnce = true,
+                legacyName = "org.jitsi.videobridge.TCP_HARVESTER_PORT",
+                newName = "videobridge.ice.tcp.port"
+            )
+            private val tcpPortProperty = TcpPortProperty()
+
+            /**
+             * The property which configures the fallback ICE/TCP port.
+             */
+            class TcpFallbackPortProperty : SimpleProperty<Int>(
+                attributes {
+                    name("videobridge.ice.tcp.fallback-port")
+                    readOnce()
+                    fromConfig(JitsiConfig.newConfig)
+                }
+            )
+            private val tcpFallbackPortProperty = TcpFallbackPortProperty()
+
+            /**
+             * The list of ports to try. Fallback can be disabled by setting the fallback port to <=0.
+             */
+            @JvmStatic
+            fun tcpPortsToTry() = if (tcpFallbackPortProperty.value > 0)
+                listOf(tcpPortProperty.value, tcpFallbackPortProperty.value)
+                else listOf(tcpPortProperty.value)
+
+            /**
+             * The property which configures an additional port to advertise.
+             */
+            class TcpMappedPortProperty : LegacyFallbackConfigProperty<Int>(
+                Int::class,
+                readOnce = true,
+                legacyName = "org.jitsi.videobridge.TCP_HARVESTER_MAPPED_PORT",
+                newName = "videobridge.ice.tcp.mapped-port"
+            )
+            private val tcpMappedPortProperty = TcpMappedPortProperty()
+
+            /**
+             * Returns the additional port to advertise, or [null] if none is configured.
+             */
+            @JvmStatic
+            fun tcpMappedPort(): Int? = try {
```

Comment:
Any reason that for `tcpMappedPort` we don't set it at all if we don't want one, but for fallback we use a negative value?  It seems like either scheme would work for both?  Is it that it doesn't make sense to provide a reasonable default for `tcpMappedPort`?

Reply:
Yes, the mapped port is disabled by default, while the fallback port is 4443 by default. Do you prefer setting `mappedPort = -1` in `reference.conf`?

Reply:
No, I was thinking at some point we should add a `isPresent` or `exists` field to `ConfigProperty` that would do the try/catch work and then stuff like this could use it.  But this is fine.

Reply:
Yes, this would be nice. We could also avoid try/catch and implement it as checks on the `ConfigResult`.